File: cartracker/src/main/webapp/app/model/option/Feature.js
Ext.define("CarTracker.model.option.Feature",
{
  extend : "Ext.data.Model",
  fields : [ {
    name : "longName",
    type : "string"
  }, {
    name : "shortName",
    type : "string"
  }, {
    name : "createDate",
    type : "date",
    dateFormat : "c"
  }, {
    name : "active",
    type : "boolean",
    defaultValue : true
  }, {
    name : "id",
    type : "int",
    useNull : true,
    convert : null
  } ],
  proxy : {
    type : "direct",
    api : {
      read : "featureOptionService.read",
      create : "featureOptionService.create",
      update : "featureOptionService.update",
      destroy : "featureOptionService.destroy"
    },
    reader : {
      root : "records"
    }
  }
});

File: cartracker/src/main/java/ch/rasc/cartracker/entity/option/Feature.java
package ch.rasc.cartracker.entity.option;

import javax.persistence.Entity;

import ch.rasc.extclassgenerator.Model;

@Entity
@Model(value = "CarTracker.model.option.Feature",
		readMethod = "featureOptionService.read",
		createMethod = "featureOptionService.create",
		updateMethod = "featureOptionService.update",
		destroyMethod = "featureOptionService.destroy", paging = true)
public class Feature extends BaseOption {
	// nothing special
}
